scripts/pipeline/orthology_benchmark.py

```diff
@@ -175,6 +178,49 @@ def get_core_names(species_names: List[str], host: str, port: int, user: str) ->
     return core_names
 
 
+def get_gtf_file(core_name: str, source_dir: str, target_dir: str) -> None:
+    """Finds and copies a GTF file for a specified core db.
+
+    Note:
+        The code utilises the structure of `production/ensemblftp` or MC's personal dir
+        to speed up file search.
+
+    Args:
+        core_name: Core db name.
+        source_dir: Path to the directory containing subdirectories with GTF files.
+            `production/ensemblftp` or MC's `nobackup/release_dumps`.
+        target_dir: Path to the directory where the GTF file will be copied.
+
+    Warns:
+        UserWarning: If the GTF file was not found.
+
+    """
+    species_name = core_name.split("_core_")[0]
+    release = core_name.split("_core_")[1].split("_")[0]
+
+    parent_dir = os.path.join(source_dir, f"release-{release}")
+    gtf_dirs_patterns = [
+        os.path.join(parent_dir, "*", "gtf", species_name),
+        os.path.join(parent_dir, "gtf", species_name)  # vertebrates in `production/ensemblftp`
+    ]
+
+    gtf_file_pattern = f"{species_name.capitalize()}.*.{release}.gtf.gz"
+    for pattern in gtf_dirs_patterns:
+        try:
+            gtf_file = list(glob.glob(os.path.join(pattern, gtf_file_pattern)))[0]
+        except IndexError:
+            continue
+
+        os.makedirs(target_dir, exist_ok=True)
+        shutil.copy(gtf_file, target_dir)
+        copied = True
+
+    try:
+        copied
+    except NameError:
```

Comment:
I would avoid testing for a successful copy using a `NameError`. If a `copied` variable is defined in the global scope, there will be no `NameError` and this will fail to warn if a GTF file was not found.

Two alternative ways to test if a GTF file was found:
1. Initialise `gtf_file` to `None` before the for loop, then issue a warning `if gtf_file is None` after the for loop.
2. Create a list of `gtf_files`, gathering all matching GTF files into that list (using a for loop, or maybe a list comprehension?), and then access the first element of that list. If you catch an `IndexError` at that point, you can be sure of the need for a 'GTF not found' warning.

Reply:
So true and so obvious... Thank you for spotting it 👍